kong/plugins/log-buffering/json_serializer.lua

```diff
@@ -0,0 +1,66 @@
+-- JSON array serializer object for using the Generic Logging Buffer.
+local cjson = require("cjson")
+
+
+local json_serializer = {}
+
+
+local cjson_encode = cjson.encode
+
+
+local function add_entry(self, data)
+  if not self.encoded then
+    data = cjson_encode(data)
+  end
+  local n = #self.buffer
+  if n == 0 then
+    self.buffer[1] = "["
+  else
+    self.buffer[n+1] = ","
+  end
+  self.buffer[n+2] = data
+  self.bytes = self.bytes + #data + 1
+  return true, (n + 2) / 2
+end
+
+
+local function serialize(self)
+  local count = #self.buffer / 2
+  self.buffer[#self.buffer + 1] = "]"
+  local data = table.concat(self.buffer)
+  return data, count, #data
+end
+
+
+local function reset(self)
+  self.buffer = {}
+  self.bytes = 1
+end
+
+
+-- Serializes the given entries into a JSON array.
+-- @param raw_tree (boolean)
+-- If `encoded` is `true`, entries are assumed to be strings
+-- that already represent JSON-encoded data.
+-- If `encoded` is `false`, entries are assumed to be Lua objects
+-- that need to be encoded during serialization.
+function json_serializer.new(encoded)
+  if encoded ~= nil and type(encoded) ~= "boolean" then
+    error("arg 2 (encoded) must be boolean")
+  end
+
+  local self = {
+    buffer = {},
```

Comment:
I find troublesome the fact that the serializers have an attribute called `buffer` but at the same time they are required by a Buffer. I think the Buffer could be renamed to Queue to make it a bit less confusing.

Reply:
the Buffer objects also have a sending queue, so that's not much better. :) Renaming those to `self.output` :)